posthog-core/src/index.ts

```diff
@@ -491,7 +562,10 @@ export abstract class PostHogCore {
     const queue = this.getPersistedProperty<PostHogQueueItem[]>(PostHogPersistedProperty.Queue) || []
 
     if (!queue.length) {
-      return callback && safeSetTimeout(callback, 0)
+      callback?.()
+      // TODO: Ask Ben what timeout on the callback really does. I don't understand
```

Comment:
Hey @benjackwhite I'm a bit mystified, not sure what's going on here:

1. Things are fine when I just run `example.ts`
2. When running jest tests though, if I wait for `safeSetTimeout`, some tests hang! (You can test this by commenting out the `callback?.()` call.

I'm not quite sure why this is happening. Do you have any ideas? I tried process dumps to figure out any unknown timers / promises that might be stopping things, but didn't manage to figure this out 😅 .

Reply:
tbh I think we can remove the safeSetTimeout here as we immediately return anyways. It was only necessary for the old API where we had lots of callbacks in a loop.

Not sure why the tests have an issue though. I guess becuase there is a hanging timeout ref if you haven't run `runAllPendingTimers` but that should be covered by the unref part of the `safeSetTimeout` function...

Reply:
that unblocks, cool